packages/eslint-plugin/src/rules/no-for-in-array.ts

```diff
@@ -45,3 +46,33 @@ export default createRule({
     };
   },
 });
+
+function isArgumentsObjectType(type: ts.Type): boolean {
+  return (
+    type.getSymbol()?.escapedName === ts.escapeLeadingUnderscores('IArguments')
```

Comment:
I'm not sure if there's a better way to recognize the type of the `arguments` object. Alternatively, I can rely on AST-alone and not have this use the types at all.